pyo3-macros-backend/src/module.rs

```diff
@@ -295,8 +295,29 @@ pub fn pymodule_function_impl(mut function: syn::ItemFn) -> Result<TokenStream>
     }
     module_args.push(quote!(::std::convert::Into::into(BoundRef(module))));
 
+    let syn::ItemFn {
+        attrs, sig, block, ..
+    } = &function;
+
+    let extractors = sig.inputs.iter().filter_map(|param| {
+        if let syn::FnArg::Typed(pat_type) = param {
+            if let syn::Pat::Ident(pat_ident) = &*pat_type.pat {
+                let ident = &pat_ident.ident;
+                return Some(quote_spanned! { pat_type.span() => {
+                    let (_, e) = #pyo3_path::impl_::pymethods::inspect_type(#ident);
```

Comment:
Might need to assign the argument back out:

```suggestion
                    let (#ident, e) = #pyo3_path::impl_::pymethods::inspect_type(#ident);
```

(I think this would matter for edge cases where the module is using `Bound<'_, PyModule>` or `Py<PyModule>` for some reason, and `inspect_type` would transfer ownership rather than copy.)